### PR TITLE
chore(js): load jQuery and UI before RequireJS

### LINKED ISSUE #9024: 2.0: Move jQuery and UI before RequireJS
This would eliminate the need to create the nasty [shim around UI](https://github.com/Elgg/Elgg/blob/2.0/views/default/jquery-ui.js.php) and allow devs to use CDN versions.

### PR DESCRIPTION
**Pull request by mrclay**
Some Elgg core components (and 3rd-party extenders of elgg.js) need jQuery UI to be available immediately. Previously we loaded jQuery and UI after RequireJS, but this required adding a complex wrapper around UI because of the way it implemented AMD support.

Since there's really no benefit in loading these 3 scripts in any particular order, we now load jQuery and UI first. This means their factories don't know about AMD, they set up the global variables we need, and then we can create dummy AMD definitions for them in elgg.js.

In this configuration all three scripts are untouched by Elgg and our JS implementation doesn't need to worry about synchronous loading in an AMD environment.

Fixes #9024

### CODE REVIEW

File: views/default/elgg.js.php [hunk 1]
```diff
@@ -65,8 +65,17 @@
 elgg.security.interval = <?php echo (int)_elgg_services()->actions->getActionTokenTimeout() * 333; ?>;
 elgg.config.language = '<?php echo (empty($CONFIG->language) ? 'en' : $CONFIG->language); ?>';
 
-// We require jquery-ui because loading the file in an AMD environment doesn't automatically
-// apply its functionality to jQuery.
+// jQuery and UI must be loaded sync in 2.x but modules should depend on these AMD modules
+define('jquery', function () {
```
Comment (ewinslow) [inline views/default/elgg.js.php]: jQuery already does this, no?

Comment (mrclay) [inline views/default/elgg.js.php]: Ah yes.

Comment (mrclay) [inline views/default/elgg.js.php]: No, because we're loading it before `define` is set up.

Comment (mrclay) [inline views/default/elgg.js.php]: To be clear, what's driving all this is that some Elgg components (and probably 3rd parties) need jQuery UI loaded synchronously, before require can load them.

Currently we cheat a bit by SCRIPT loading them before elgg.js. jQuery core's module setup is flexible allowing both synchronous defining of the `jQuery` global var as well as auto-creating an AMD definition for "jquery".

jQuery UI's module setup is dumber: it just checks if `define` exists, and assumes if it does that you don't want the global `jQuery` var extended.

Comment (mrclay) [inline views/default/elgg.js.php]: In this PR, jQuery and UI just set up standard browser globals (which we still need) and then we continue on with Require and elgg.js. 3.x plugins still won't be able to use jQuery or UI before the HTML footer, so I think this is still successful in pushing devs towards AMD.

Comment (mrclay) [inline views/default/elgg.js.php]: I'll summarize this better in the commit msg...

**Comment by mrclay** [comment]
I think this is important to consider, putting it on the milestone.

**Comment by mrclay** [comment]
Pulling this in. Hubris!

**Pull request merged**